pkg/controller/perconaservermongodb/status.go

```diff
@@ -92,14 +94,60 @@ func (r *ReconcilePerconaServerMongoDB) updateStatus(cr *api.PerconaServerMongoD
 		}
 	}
 
+	matchLabels := map[string]string{
+		"app.kubernetes.io/name":       "percona-server-mongodb",
+		"app.kubernetes.io/instance":   cr.Name,
+		"app.kubernetes.io/replset":    cr.Spec.Replsets[0].Name,
+		"app.kubernetes.io/managed-by": "percona-server-mongodb-operator",
+		"app.kubernetes.io/part-of":    "percona-server-mongodb",
+	}
+
+	pods := &corev1.PodList{}
+	err := r.client.List(context.TODO(),
+		&client.ListOptions{
+			Namespace:     cr.Namespace,
+			LabelSelector: labels.SelectorFromSet(matchLabels),
+		},
+		pods,
+	)
+	if err != nil {
+		return r.writeStatus(cr)
+	}
+	secrets := &corev1.Secret{}
+	err = r.client.Get(
+		context.TODO(),
+		types.NamespacedName{Name: cr.Spec.Secrets.Users, Namespace: cr.Namespace},
+		secrets,
+	)
+	if err != nil {
+		err = errors.Wrap(err, "get mongodb secrets")
+		return r.writeStatus(cr)
+	}
+	rsAddrs, err := psmdb.GetReplsetAddrs(r.client, cr, cr.Spec.Replsets[0], pods.Items)
+	session, err := mongo.Dial(rsAddrs, cr.Spec.Replsets[0].Name, secrets, true)
+	if err != nil {
+		session, err = mongo.Dial(rsAddrs, cr.Spec.Replsets[0].Name, secrets, false)
+		if err != nil {
+			return r.writeStatus(cr)
+		}
+	}
```

Comment:
I'd like to avoid extra connections since it gonna be running on each reconcile cycle. We already have that stuff in https://github.com/percona/percona-server-mongodb-operator/blob/a664d0f586f12b4fed893fdb6597b727b340a48c/pkg/controller/perconaservermongodb/mgo.go#L19
I suggest changing `reconcileCluster()` to return also the number of live nodes. And passing that "liveNodesNum" as the parameter here (`updateStatus()`)